lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -0,0 +1,132 @@
+            $result.ansible_facts.ansible_disk."disk_$($i)"["physical_disk)"].spindle_speed = "$($pdisk.SpindleSpeed)rpm"
+            $result.ansible_facts.ansible_disk."disk_$($i)"["physical_disk)"].can_pool = $pdisk.CanPool
+            if (-not $pdisk.CanPool) {
+                $result.ansible_facts.ansible_disk."disk_$($i)"["physical_disk)"].cannot_pool_reason = $pdisk.CannotPoolReason
+            }   
+            $result.ansible_facts.ansible_disk."disk_$($i)"["physical_disk)"].object_id = $pdisk.ObjectId
+            $result.ansible_facts.ansible_disk."disk_$($i)"["physical_disk)"].unique_id = $pdisk.UniqueId
+            $vdisk = Get-VirtualDisk -PhysicalDisk $pdisk -ErrorAction SilentlyContinue
+            if ($vdisk) {
+                $result.ansible_facts.ansible_disk["disk_$($i)"]["virtual_disk)"] += @{}
+                $result.ansible_facts.ansible_disk."disk_$($i)"["virtual_disk)"].friendly_name = $vdisk.FriendlyName
+            }
+        }
+        $result.ansible_facts.ansible_disk."disk_$($i)".number = $disks.Number
+        $result.ansible_facts.ansible_disk."disk_$($i)".size = "$($disks.Size / 1GB)gb"
+        $result.ansible_facts.ansible_disk."disk_$($i)".bus_type = $disks.BusType
+        $result.ansible_facts.ansible_disk."disk_$($i)".friendly_name = $disks.FriendlyName
+        $result.ansible_facts.ansible_disk."disk_$($i)".partition_style = $disks.PartitionStyle
+        $result.ansible_facts.ansible_disk."disk_$($i)".partition_count = $disks.NumberOfPartitions
+        $result.ansible_facts.ansible_disk."disk_$($i)".operational_status = $disks.OperationalStatus
+        $result.ansible_facts.ansible_disk."disk_$($i)".sector_size = "$($disks.PhysicalSectorSize)byte"
+        $result.ansible_facts.ansible_disk."disk_$($i)".read_only = $disks.IsReadOnly
+        $result.ansible_facts.ansible_disk."disk_$($i)".bootable = $disks.IsBoot
+        $result.ansible_facts.ansible_disk."disk_$($i)".system_disk = $disks.IsSystem
+        $result.ansible_facts.ansible_disk."disk_$($i)".clustered = $disks.IsClustered
+        $result.ansible_facts.ansible_disk."disk_$($i)".manufacturer = $disks.Manufacturer
+        $result.ansible_facts.ansible_disk."disk_$($i)".model = $disks.Model
+        $result.ansible_facts.ansible_disk."disk_$($i)".firmware_version = $disks.FirmwareVersion
+        $result.ansible_facts.ansible_disk."disk_$($i)".location = $disks.Location
+        $result.ansible_facts.ansible_disk."disk_$($i)".serial_number = $disks.SerialNumber
+        $result.ansible_facts.ansible_disk."disk_$($i)".unique_id = $disks.UniqueId
+        $result.ansible_facts.ansible_disk."disk_$($i)".guid = "$([string]$disks.Guid)"
+        $result.ansible_facts.ansible_disk."disk_$($i)".path = $disks.Path
+        $part = Get-Partition -DiskNumber $($disks.Number) -ErrorAction SilentlyContinue
+        if ($part) {
+            $j = 1
```

Comment:
Same for partitions, each `$disk_result` should be a list of partitions and not a dict with arbitrary keys.

Reply:
same as for disks